hs_core/models.py

```diff
@@ -1668,6 +1666,33 @@ class MyResourceContentType(pages.Page, hs_core.AbstractResource):
     # this field WILL NOT get recorded in bag and SHOULD NEVER be used for storing metadata
     extra_data = HStoreField(default={})
 
+    @property
+    def view_count(self):
+        return int(self.extra_data.get('view_count', '0'))
+
+    @property
+    def download_count(self):
+        return int(self.extra_data.get('download_count', '0'))
+
+    def update_view_count(self, request):
+        resource_ids = request.session.get('resource_ids', [])
+        if resource_ids:
+            if self.short_id not in resource_ids:
+                self.__update_count('view_count')
+                request.session['resource_ids'].append(self.short_id)
+        else:
+            self.__update_count('view_count')
+            request.session['resource_ids'] = [self.short_id]
+
+    def update_download_count(self):
+        self.__update_count('download_count')
+
+    def __update_count(self, key):
+        cur_count = self.extra_data.get(key, '0')
+        new_count = int(cur_count) + 1
+        self.extra_data[key] = str(new_count)
+        self.save()
```

Comment:
Why did you decide to use extra_data for these counters?  Looking at extra_data, we're not actually using it anywhere up until now and I'm not sure this is the best use for it.  The view count is arguably the most updated field on a resource.  I think it deserves it's own field for quicker access/update.  I get that it's not a huge difference in overall time, but I'd expect it takes a whole lot more time to access the extra_data field, look up the key, convert it to integer, increment the integer, convert it back to string, insert into the map and save the field, than it would be to just increment an integer field.

Reply:
I agree that using model level individual fields for the counters would be simpler. Initially I was planning to do that. Then I thought there may be need for additional such counters in the future - leading to model level modifications and migrations. The 'extra_data' field has been used by collection resource type. I will make these counters model level fields.